addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js

```diff
@@ -236,13 +233,16 @@ export class PivotCorePlugin extends OdooCorePlugin {
     /**
      * @param {string} id
      * @param {PivotDefinition} pivot
+     * @param {string} formulaId
      */
-    _addPivot(id, pivot) {
+    _addPivot(id, pivot, formulaId = this.nextFormulaId.toString()) {
         const pivots = { ...this.pivots };
-        pivots[id] = pivot;
+        pivots[id] = {
+            ...pivot,
+            formulaId,
```

Comment:
while we are at it, it can be simplified without copying the object.
`this.history.update("pivots", id, { ...pivot, formulaId })`